LunaDll/GlobalFuncs.cpp

```diff
@@ -26,6 +27,8 @@
 #include "SMBXInternal/NPCs.h"
 #include "Misc/RuntimeHook.h"
 
+static std::mutex g_editorIPCMutex;
```

Comment:
This is not correct. The purpose of using a mutex is to prevent `GetEditorPlacedItem()` and `IPCSendItemPlacing()` accessing the `gEditorPlacedItem` variable at the same time. For the locks to do this, the locks must refer to _the same mutex object_. Note that statically declaring it in two different files creates two _separate_ mutex objects, so it won't do it's job at all. Furthermore static variables are _not_ global as the prefix you put in the name might imply. You could be making the mutex a global variable in the same way you made gEditorPlacedItem a global variable, for the exact same reasons.